qiskit/qpy/binary_io/schedules.py

```diff
@@ -60,6 +61,22 @@ def _read_waveform(file_obj, version):
     )
 
 
+def _read_kernel_and_discriminator(file_obj, version, kernel_or_discriminator):
```

Comment:
Unless I am missing a technical reason to have these two objects handled in the same function, I think I would have gone for two separate functions.